fuzz/fuzz_targets/encode_decode.rs

```diff
@@ -33,12 +33,17 @@ fn do_test(data: &[u8]) {
             match Hrp::parse(&s) {
                 Err(_) => return,
                 Ok(hrp) => {
-                    if let Ok(data_str) = bech32::encode(hrp, &dp, variant).map(|b32| b32.to_string()) {
-                        let decoded = bech32::decode(&data_str);
-                        let b32 = decoded.expect("should be able to decode own encoding");
+                    let encoded = bech32::encode(hrp, &dp, variant).expect("failed to encode");
+                    let decoded = bech32::decode(&encoded);
+                    let (parsed, variant) = match decoded {
+                        Ok((parsed, variant)) => (parsed, variant),
+                        Err(_) => return,
```

Comment:
In e2527dc19c3e8e9a4cd678a955fc5b161e175c87:

Decoding should never fail, if this came from the encoding of something we just parsed, right?

Reply:
Yes, you are correct. Used `expect` instead. Thanks